src/App.tsx

```diff
@@ -1,61 +1,193 @@
+    ]);
+
+    nextTodoId += 1;
+  };
+
+  const handleSubmit = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+
+    const isEachFieldValid = validateFields();
+
+    if (isEachFieldValid) {
+      addNewTodo();
+
+      setTitle('');
+      setUserId(0);
+    }
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value: newString } = event.target;
+
+    setTitle(newString.replace(/[^0-9a-zа-яіїєґ\s]/gi, ''));
+    setIsTitleErrorShown(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(Number(event.target.value));
+    setIsUserIdErrorShown(false);
+  };
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/users" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      <div className="container">
+        <h1 className="title">Add todo form</h1>
+
+        <form
+          action="/api/users"
+          method="POST"
+          onSubmit={handleSubmit}
+        >
+          <div className="field">
+            <div className="control">
+              <label
+                htmlFor="todoTitle"
+                className="label"
+              >
+                {'Title: '}
+                <input
+                  type="text"
+                  name="title"
+                  className={classNames(
+                    'input',
+                    {
+                      'is-danger': isTitleErrorShown,
+                    },
+                  )}
+                  id="todoTitle"
```

Comment:
do u need this `id`?

Reply:
No I don't. But I think it's better to explicitly set `id` attribute for the `input` and the `for` attribute for the `label`, but I have no strong arguments why. So why not?